mergify_engine/actions/draft.py

```diff
@@ -0,0 +1,108 @@
+from mergify_engine.rules import types
+
+
+class DraftAction(actions.Action):
+    flags = (
+        actions.ActionFlag.ALLOW_AS_ACTION
+        | actions.ActionFlag.ALWAYS_RUN
+        | actions.ActionFlag.ALLOW_ON_CONFIGURATION_CHANGED
+    )
+
+    validator = {
+        voluptuous.Required("bot_account", default=None): types.Jinja2WithNone,
+    }
+
+    async def run(
+        self, ctxt: context.Context, rule: rules.EvaluatedRule
+    ) -> check_api.Result:
+        if ctxt.pull["draft"]:
+            return check_api.Result(
+                check_api.Conclusion.SUCCESS,
+                "Pull Request was already converted to draft.",
+                "",
+            )
+        try:
+            bot_account = await action_utils.render_bot_account(
+                ctxt,
+                self.config["bot_account"],
+                required_feature=subscription.Features.BOT_ACCOUNT,
+                missing_feature_message="Draft with `bot_account` set are disabled",
+                required_permissions=[],
+            )
+        except action_utils.RenderBotAccountFailure as e:
+            return check_api.Result(e.status, e.title, e.reason)
+        tokens = await user_tokens.UserTokens.select_users_for(ctxt, bot_account)
+        if not tokens or tokens[0]["oauth_access_token"] is None:
+            return check_api.Result(
+                check_api.Conclusion.FAILURE,
+                f"Unable to set draft with user `{tokens[0]['login']}`",
+                f"Please make sure `{tokens[0]['login']}` has logged in Mergify dashboard.",
+            )
+        mutation = f"""
+            mutation {{
+                convertPullRequestToDraft(input:{{pullRequestId: "{ctxt.pull['node_id']}"}}) {{
+                    pullRequest {{
+                        isDraft
+                    }}
+                }}
+            }}
+        """
+        response = (
+            await ctxt.client.post(
+                "/graphql",
+                json={"query": mutation},
+                oauth_token=tokens[0]["oauth_access_token"],
+            )
+        ).json()
+        if response["data"] is None:
+            ctxt.log.error(
+                "Graphql API call failed; unable to convert PR to draft.",
+                response=response,
+            )
+            return check_api.Result(
+                check_api.Conclusion.FAILURE,
+                "GraphQL API call failed; pull request wasn't converted to draft.",
+                "",
+            )
+        if response["data"]["convertPullRequestToDraft"]["pullRequest"]["isDraft"]:
+            ctxt.pull["draft"] = True
+            return check_api.Result(
+                check_api.Conclusion.SUCCESS,
+                "Pull request successfully converted to draft",
+                "",
+            )
+        return check_api.Result(
+            check_api.Conclusion.FAILURE,
+            "Pull request wasn't converted to draft.",
+            "",
+        )
```

Comment:
@sileht Do we really want to check this? If we return an error to the user and don't even log it, I'm not sure how we can be aware of it. I also don't think we can fix it since it's likely a GitHub bug. So is it worth it to check for github bugs?

Reply:
> @sileht Do we really want to check this? If we return an error to the user and don't even log it, I'm not sure how we can be aware of it. I also don't think we can fix it since it's likely a GitHub bug. So is it worth it to check for github bugs?

Agreed, the earlier checks is enought:

```suggestion
        ctxt.pull["draft"] = True
        return check_api.Result(
            check_api.Conclusion.SUCCESS,
            "Pull request successfully converted to draft",
            "",
        )
```